homeassistant/components/rympro/coordinator.py

```diff
@@ -0,0 +1,38 @@
+"""The Read Your Meter Pro integration."""
+from __future__ import annotations
+
+from datetime import timedelta
+import logging
+
+from pyrympro import CannotConnectError, OperationError, RymPro, UnauthorizedError
+
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator, UpdateFailed
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class RymProDataUpdateCoordinator(DataUpdateCoordinator):
+    """Class to manage fetching RYM Pro data."""
+
+    def __init__(self, hass: HomeAssistant, rympro: RymPro, scan_interval: int) -> None:
+        """Initialize global RymPro data updater."""
+        self.rympro = rympro
+        interval = timedelta(seconds=scan_interval)
+        super().__init__(
+            hass,
+            _LOGGER,
+            name=DOMAIN,
+            update_interval=interval,
+        )
+
+    async def _async_update_data(self):
+        """Fetch data from Rym Pro."""
+        try:
+            return await self.rympro.last_read()
+        except UnauthorizedError:
+            await self.hass.config_entries.async_reload(self.config_entry.entry_id)
```

Comment:
```suggestion
        except UnauthorizedError as error:
            assert self.config_entry
            await self.hass.config_entries.async_reload(self.config_entry.entry_id)
            raise UpdateFailed(error) from error
```

Reply:
I'm not sure we should raise `UpdateFailed` after handling this by reloading the config entry.

Reply:
I think since this trigger reload and reload will pull try to login again and pull the data immediately it would actually be better not to raise `UpdateFailed` since we don't want to notify the user with an error if there is no action needed from his side. If login fails we raise anyhow after reload.

Reply:
The problem is that we need to return something now, and I'm not sure what :)